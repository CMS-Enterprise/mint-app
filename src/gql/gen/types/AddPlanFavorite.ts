/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: AddPlanFavorite
// ====================================================

export interface AddPlanFavorite_addPlanFavorite {
  __typename: "PlanFavorite";
  modelPlanID: UUID;
  userID: UUID;
}

export interface AddPlanFavorite {
  addPlanFavorite: AddPlanFavorite_addPlanFavorite;
}

export interface AddPlanFavoriteVariables {
  modelPlanID: UUID;
}
