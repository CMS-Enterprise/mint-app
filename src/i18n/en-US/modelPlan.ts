const modelPlan = {
  home: 'Home',
  teamRoles: {
    modelLead: 'Model Lead',
    leadership: 'Leadership',
    evaluation: 'Evaluation',
    learning: 'Learning',
    modelTeam: 'Model Team',
    itLead: 'IT Lead',
    quality: 'Quality'
  },
  planStatuses: {
    planDraft: 'Draft model plan',
    planComplete: 'Model plan complete',
    icipComplete: 'ICIP complete',
    cmmiClearance: 'Internal (CMMI) clearance',
    cmsClearance: 'CMS clearance',
    hhsClearance: 'HHS clearance',
    ombASRFClearance: 'OMB/ASRF clearance',
    cleared: 'Cleared',
    announced: 'Announced',
    paused: 'Paused',
    canceled: 'Canceled'
  },
  status: {
    heading: 'Update status',
    copy:
      'After you’ve iterated on your Model Plan, update the status so others know what stage it’s at in the design and clearance process.',
    label: 'What is the status of your Model Plan?',
    updateButton: 'Update status',
    return: 'Don’t update status and return to task list'
  },
  favorite: {
    modelLead: 'Model lead(s)',
    startDate: 'Start date',
    cRTDLs: 'CRs and TDLs',
    toBeDetermined: 'To be determined',
    noneEntered: 'None entered',
    follow: 'Follow',
    following: 'Following',
    success:
      'Success! You are no longer following {{-requestName}} and will not receive email notifications about this model.',
    failure:
      'There was an error unfollowing {{-requestName}}. Please try unfollowing the model again if you no longer wish to receive email notifications about it.',
    error: 'Failed to locate and unfollow model plan.'
  }
};

export default modelPlan;
