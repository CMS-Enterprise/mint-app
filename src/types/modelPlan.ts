import { TaskStatus } from './graphql-global-types';

export enum CLEARANCE_STATUS {
  CANNOT_START = 'CANNOT_START'
}

export type ModelPlanTaskStatus = TaskStatus | CLEARANCE_STATUS;
