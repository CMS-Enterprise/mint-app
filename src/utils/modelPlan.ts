import i18next from 'i18next';
/**
 * Translate the API enum to a human readable string
 */
const translateTeamRole = (teamRole: string) => {
  switch (teamRole) {
    case 'MODEL_LEAD':
      return i18next.t('modelPlan:teamRoles.modelLead');
    default:
      return '';
  }
};

export default translateTeamRole;
