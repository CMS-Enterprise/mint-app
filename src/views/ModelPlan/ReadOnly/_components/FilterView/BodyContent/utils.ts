const allPossibleFilterViews = {
  mdm: {
    basics: ['nameHistory'],
    beneficiaries: [
      'beneficiaries',
      'beneficiariesOther',
      'beneficiariesNote',
      'numberPeopleImpacted',
      'estimateConfidence',
      'beneficiaryOverlap',
      'beneficiaryOverlapNote',
      'precedenceRules'
    ]
  },
  filler: {
    basics: ['nameHistory']
  }
};

export default allPossibleFilterViews;
