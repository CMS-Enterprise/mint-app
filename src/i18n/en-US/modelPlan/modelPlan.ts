import { ModelViewFilter } from 'gql/gen/graphql';

import { TranslationModelPlan } from 'types/translation';

export const modelPlan: TranslationModelPlan = {
  modelName: {
    gqlField: 'modelName',
    goField: 'ModelName',
    dbField: 'model_name',
    label: 'Model name',
    dataType: 'string',
    formType: 'text'
  },
  previousName: {
    gqlField: 'previousName',
    goField: 'PreviousName',
    dbField: 'previous_name',
    label: 'Previous names',
    dataType: 'string',
    formType: 'text'
  },
  nameHistory: {
    gqlField: 'nameHistory',
    goField: 'NameHistory',
    dbField: 'name_history',
    label: 'Previous names',
    dataType: 'string',
    isArray: true,
    formType: 'text',
    filterGroups: [
      ModelViewFilter.OACT,
      ModelViewFilter.DFSDM,
      ModelViewFilter.CCW,
      ModelViewFilter.IPC,
      ModelViewFilter.IDDOC,
      ModelViewFilter.PBG,
      ModelViewFilter.MDM,
      ModelViewFilter.CBOSC
    ]
  },
  abbreviation: {
    gqlField: 'abbreviation',
    goField: 'Abbreviation',
    dbField: 'abbreviation',
    label: 'Short name',
    sublabel:
      'The abbreviation, acronym, or other common name used for the model.',
    dataType: 'string',
    formType: 'text'
  },
  archived: {
    gqlField: 'archived',
    goField: 'Archived',
    dbField: 'archived',
    label: 'Archived',
    dataType: 'boolean',
    formType: 'radio',
    options: {
      true: 'Yes',
      false: 'No'
    }
  },
  status: {
    gqlField: 'status',
    goField: 'Status',
    dbField: 'status',
    label: 'What is the status of your Model Plan?',
    dataType: 'enum',
    formType: 'select',
    options: {
      PLAN_DRAFT: 'Draft Model Plan',
      PLAN_COMPLETE: 'Model Plan complete',
      ICIP_COMPLETE: 'ICIP complete',
      INTERNAL_CMMI_CLEARANCE: 'In internal (CMMI) clearance',
      CMS_CLEARANCE: 'In CMS clearance',
      HHS_CLEARANCE: 'In HHS clearance',
      OMB_ASRF_CLEARANCE: 'In OMB clearance',
      CLEARED: 'Cleared',
      ANNOUNCED: 'Announced',
      ACTIVE: 'Active',
      ENDED: 'Ended',
      PAUSED: 'Paused',
      CANCELED: 'Canceled'
    }
  }
};

export const modelPlanMisc: Record<string, string> = {
  heading: 'Model name',
  modelID: 'Model ID',
  createdBy: 'Created by',
  createdAt: 'Created at',
  readyForReviewBy: 'Ready for review by',
  readyForReviewAt: 'Ready for review at',
  breadcrumb: 'Start a new Model Plan',
  modeName: 'What is the name of your model?',
  modelNameInfo:
    'This is not a permanent name. If needed, you may update it later.',
  headingStatus: 'Update status',
  copy:
    'After you’ve iterated on your Model Plan, update the status so others know what stage it’s at in the design and clearance process.',
  updateButton: 'Update status',
  return: 'Don’t update status and return to task list'
};

export default modelPlan;
