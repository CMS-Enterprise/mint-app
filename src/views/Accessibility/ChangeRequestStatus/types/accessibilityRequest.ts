/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AccessibilityRequestStatus } from "./../../../../types/graphql-global-types";

// ====================================================
// GraphQL query operation: accessibilityRequest
// ====================================================

export interface accessibilityRequest_accessibilityRequest_statusRecord {
  __typename: "AccessibilityRequestStatusRecord";
  status: AccessibilityRequestStatus;
}

export interface accessibilityRequest_accessibilityRequest {
  __typename: "AccessibilityRequest";
  id: UUID;
  name: string;
  statusRecord: accessibilityRequest_accessibilityRequest_statusRecord;
}

export interface accessibilityRequest {
  accessibilityRequest: accessibilityRequest_accessibilityRequest | null;
}

export interface accessibilityRequestVariables {
  id: UUID;
}
