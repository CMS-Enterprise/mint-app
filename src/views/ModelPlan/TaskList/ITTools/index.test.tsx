import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { Formik } from 'formik';

import { GcPartCDType } from 'types/graphql-global-types';
import { translateGcPartCDType } from 'utils/modelPlan';

import { ITToolsFormComponent } from '.';

const onSubmit = (values: any) => {};

const formikValues = {
  gcPartCD: ['MARX', 'OTHER'],
  gcPartCDOther: 'Test Other'
};

describe('The ITToolsFormComponent component', () => {
  it('renders checkbox options if needs tool', async () => {
    const { getByTestId } = render(
      <Formik initialValues={formikValues} onSubmit={onSubmit}>
        <ITToolsFormComponent
          flatErrors={{}}
          formikValue={formikValues.gcPartCD}
          fieldName="gcPartCD"
          needsTool
          htmlID="gc-partc"
          EnumType={GcPartCDType}
          translation={translateGcPartCDType}
        />
      </Formik>
    );

    await waitFor(() => {
      expect(getByTestId('it-tools-gc-partc-MARX')).toBeChecked();
      expect(getByTestId('it-tools-gc-partc-OTHER')).toBeChecked();
      expect(getByTestId('it-tools-gc-partc-other')).toHaveValue('Test Other');
    });
  });

  it('renders checkbox options if does not need tool', async () => {
    const { getByTestId } = render(
      <Formik initialValues={{ gcPartCD: [] }} onSubmit={onSubmit}>
        <ITToolsFormComponent
          flatErrors={{}}
          formikValue={formikValues.gcPartCD}
          fieldName="gcPartCD"
          needsTool={false}
          htmlID="gc-partc"
          EnumType={GcPartCDType}
          translation={translateGcPartCDType}
        />
      </Formik>
    );

    await waitFor(() => {
      expect(getByTestId('it-tools-gc-partc-MARX')).toHaveAttribute('disabled');
      expect(getByTestId('it-tools-gc-partc-OTHER')).toHaveAttribute(
        'disabled'
      );
      expect(() => getByTestId('it-tools-gc-partc-other')).toThrow();
    });
  });

  it('matches snapshot', async () => {
    const { asFragment } = render(
      <Formik initialValues={formikValues} onSubmit={onSubmit}>
        <ITToolsFormComponent
          flatErrors={{}}
          formikValue={formikValues.gcPartCD}
          fieldName="gcPartCD"
          needsTool
          htmlID="gc-partc"
          EnumType={GcPartCDType}
          translation={translateGcPartCDType}
        />
      </Formik>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
