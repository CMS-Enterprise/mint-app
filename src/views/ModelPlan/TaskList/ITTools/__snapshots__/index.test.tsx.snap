// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ITToolsFormComponent component matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="margin-top-4"
  >
    Tools
  </p>
  <div
    class="mint-checkbox usa-checkbox"
  >
    <input
      checked=""
      class="usa-checkbox__input"
      data-testid="it-tools-gc-partc-MARX"
      id="it-tools-gc-partc-MARX"
      name="gcPartCD"
      type="checkbox"
      value="MARX"
    />
    <label
      class="usa-checkbox__label"
      for="it-tools-gc-partc-MARX"
    >
      Medicare Advantage Prescription Drug System (MARx)
    </label>
    <p
      class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
    />
  </div>
  <div
    class="mint-checkbox usa-checkbox"
  >
    <input
      checked=""
      class="usa-checkbox__input"
      data-testid="it-tools-gc-partc-OTHER"
      id="it-tools-gc-partc-OTHER"
      name="gcPartCD"
      type="checkbox"
      value="OTHER"
    />
    <label
      class="usa-checkbox__label"
      for="it-tools-gc-partc-OTHER"
    >
      Other
    </label>
    <p
      class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs"
    />
  </div>
  <div
    class="margin-left-4 margin-top-1"
  >
    <label
      class="usa-label text-normal"
      data-testid="label"
      for="it-tools-gc-partc-other"
    >
      Please specify
    </label>
    <input
      class="usa-input maxw-none"
      data-testid="it-tools-gc-partc-other"
      id="it-tools-gc-partc-other"
      maxlength="50"
      name="gcPartCDOther"
      type="text"
      value="Test Other"
    />
  </div>
  <div
    class="margin-top-4 margin-bottom-8"
  >
    <button
      class="usa-button usa-button usa-button--unstyled"
      data-testid="add-note-toggle"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="usa-icon margin-right-1"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      Add an additional note
    </button>
  </div>
</DocumentFragment>
`;
